src/connectors/TokenConnectorLogic.sol

```diff
@@ -26,40 +30,18 @@ abstract contract TokenConnectorLogic is IBridgeConnector {
     event ClaimAccrued(address indexed account, uint256 value);
     event Claimed(address indexed account, uint256 value);
 
-    /**
-     * @dev Refunds the specified amount to the given receiver.
-     * @param receiver The address of the receiver.
-     * @param amount The amount to be refunded.
-     */
-    function refund(address payable receiver, uint256 amount) public override {
-        (bool refundSuccess,) = receiver.call{value: amount}("");
-        if (!refundSuccess) {
-            revert RefundFailed({recipient: receiver, amount: amount});
+     modifier onlySettled(uint256 lockAmount, bool isNative) {
```

Comment:
Won't it be better to make separate modifiers?

Reply:
Until this is understandable I don't think we'd need it cuz the only thing we need or not need to do is an addition